<?xml version="1.0"?>
<robot name="gofa_arm_urdf">

    <material name="gray_dark">
        <color rgba="0.7 0.7 0.7 1.0"/>
    </material>


    <link name="world"/>

    <link name="base_link">
        <visual>
            <geometry>
                <mesh filename="package://pastabot_pkg/mesh/GoFa_Base.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
            <material name="gray_dark"/>I            
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://pastabot_pkg/mesh/GoFa_Base.stl" scale="0.001 0.001 0.001"/>
            </geometry>
        </collision>
        <inertial>
            <mass value="10"/>
            <inertia ixx="0.4" ixy="0.0" ixz="0.0" iyy="0.4" iyz="0.0" izz="0.2"/>
        </inertial>

    </link>

    <link name="link01_rev1">
        <visual>
            <geometry>
                <mesh filename="package://pastabot_pkg/mesh/GoFa_Link1.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
            <material name="gray_dark"/>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://pastabot_pkg/mesh/GoFa_Link1.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
        </collision>
        
    </link>

    <link name="link01_DH">
        <inertial>
            <mass value="1"/>
            <!--inertia ixx="0.4" ixy="0.0" ixz="0.0" iyy="0.4" iyz="0.0" izz="0.2"/-->
            <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
        </inertial>
    </link>

    <link name="link02_rev2">
        <visual>
            <geometry>
                    <mesh filename="package://pastabot_pkg/mesh/GoFa_Link2.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="0.0 0.265 0.0" rpy="1.570795 0.0 0.0"/>
            <material name="gray_dark"/>
        </visual>
        <collision>
            <geometry>
                    <mesh filename="package://pastabot_pkg/mesh/GoFa_Link2.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="0.0 0.265 0.0" rpy="1.570795 0.0 0.0"/>
        </collision>
        
    </link>

    <link name="link02_DH">
        <inertial>
            <mass value="1"/>
            <!--inertia ixx="0.4" ixy="0.0" ixz="0.0" iyy="0.4" iyz="0.0" izz="0.2"/-->
            <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
        </inertial>
    </link>

    <link name="link03_rev3">
        <visual>
            <geometry>
                    <mesh filename="package://pastabot_pkg/mesh/GoFa_Link3.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="-0.709 0.0 0.0 " rpy="1.570795 0.0 1.570795"/>
            <material name="gray_dark"/>
        </visual>
        <collision>
            <geometry>
                    <mesh filename="package://pastabot_pkg/mesh/GoFa_Link3.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="-0.709 0.0 0.0 " rpy="1.570795 0.0 1.570795"/>
        </collision>
        
    </link>

    <link name="link03_DH">
        <inertial>
            <mass value="1"/>
            <!--inertia ixx="0.4" ixy="0.0" ixz="0.0" iyy="0.4" iyz="0.0" izz="0.2"/-->
            <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
        </inertial>
    </link>


    <link name="link04_rev4">
        <visual>
            <geometry>
                <mesh filename="package://pastabot_pkg/mesh/GoFa_Link4.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="-0.819 0.0 0.0" rpy="3.14159 -1.570795 0.0"/>
            <material name="gray_dark"/>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://pastabot_pkg/mesh/GoFa_Link4.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="-0.819 0.0 0.0" rpy="3.14159 -1.570795 0.0"/>
        </collision>
        
    </link>

    <link name="link04_DH">
        <inertial>
            <mass value="1"/>
            <!--inertia ixx="0.4" ixy="0.0" ixz="0.0" iyy="0.4" iyz="0.0" izz="0.2"/-->
            <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
        </inertial>
    </link>


    <link name="link05_rev5">
        <visual>
            <geometry>
                <mesh filename="package://pastabot_pkg/mesh/GoFa_Link5.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="-0.819 -0.47 0.0" rpy="1.570795 0.0 1.570795"/>
            <material name="gray_dark"/>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://pastabot_pkg/mesh/GoFa_Link5.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="-0.819 -0.47 0.0" rpy="1.570795 0.0 1.570795"/>
        </collision>
        
    </link>

    <link name="link05_DH">
        <inertial>
            <mass value="1"/>
            <!--inertia ixx="0.4" ixy="0.0" ixz="0.0" iyy="0.4" iyz="0.0" izz="0.2"/-->
            <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
        </inertial>
    </link>


    <link name="link06_rev6">
        <visual>
            <geometry>
                <mesh filename="package://pastabot_pkg/mesh/GoFa_Link6.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="-0.899 0.0 -0.47" rpy="0.0 -1.570795 3.14159"/>
            <material name="gray_dark"/>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://pastabot_pkg/mesh/GoFa_Link6.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="-0.899 0.0 -0.47" rpy="0.0 -1.570795 3.14159"/>
            <material name="gray_dark"/>
        </collision>
        

    </link>

    <link name="link06_DH">
        <inertial>
            <mass value="1"/>
            <!--inertia ixx="0.4" ixy="0.0" ixz="0.0" iyy="0.4" iyz="0.0" izz="0.2"/-->
            <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
        </inertial>
    </link>
    
    <link name="gripper_fixed">
        <visual>
            <geometry>
            

                <mesh filename="package://pastabot_pkg/mesh/SR24_02_ExeAHTool.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="0 0.0 -0.150" rpy="0.0 0 0"/>
            <material name="orange"/>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://pastabot_pkg/mesh/SR24_02_ExeAHTool.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <origin xyz="0 0.0 -0.150" rpy="0.0 0 0"/>
        </collision>
        <inertial>
            <mass value="1"/>
            <!--inertia ixx="0.4" ixy="0.0" ixz="0.0" iyy="0.4" iyz="0.0" izz="0.2"/-->
            <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
        </inertial>
    </link>


   
    <joint name="fixed" type="fixed">
        <parent link="world" />
        <child link="base_link" />
    </joint>

    <joint name="jnt1_rev" type="revolute">
        <parent link="base_link"/>
        <child link="link01_rev1"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>

    <joint name="jnt1_DH" type="fixed">
        <parent link="link01_rev1"/>
        <child link="link01_DH"/>
        <origin xyz="0 0 0.265" rpy="-1.570795 0 0"/>
    </joint>

    <joint name="jnt2_rev" type="revolute">
        <parent link="link01_DH"/>
        <child link="link02_rev2"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>
        
    <joint name="jnt2_DH" type="fixed">
        <parent link="link02_rev2"/>
        <child link="link02_DH"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 -0.444 0" rpy="0 0 -1.570795"/>
    </joint>

    <joint name="jnt3_rev" type="revolute">
        <parent link="link02_DH"/>
        <child link="link03_rev3"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>
        
    <joint name="jnt3_DH" type="fixed">
        <parent link="link03_rev3"/>
        <child link="link03_DH"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0.11 0 0" rpy="-1.570795 0 0"/>
    </joint>


    <joint name="jnt4_rev" type="revolute">
        <parent link="link03_DH"/>
        <child link="link04_rev4"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>
        
    <joint name="jnt4_DH" type="fixed">
        <parent link="link04_rev4"/>
        <child link="link04_DH"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0.47" rpy="1.570795 0 0"/>
    </joint>


    <joint name="jnt5_rev" type="revolute">
        <parent link="link04_DH"/>
        <child link="link05_rev5"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>
        
    <joint name="jnt5_DH" type="fixed">
        <parent link="link05_rev5"/>
        <child link="link05_DH"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0.08 0 0" rpy="-1.570795 0 0"/>
    </joint>


    <joint name="jnt6_rev" type="revolute">
        <parent link="link05_DH"/>
        <child link="link06_rev6"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>
        
    <joint name="jnt6_DH" type="fixed">
        <parent link="link06_rev6"/>
        <child link="link06_DH"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0.101" rpy="0 0 3.14159"/>
    </joint>
    
    <joint name="jnt_gripper" type="fixed">
        <parent link="link06_DH"/>
        <child link="gripper_fixed"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0.150" rpy="0 0 0"/>
    </joint>

   
    <material name="orange">
        <color rgba="0.43137254901960786 0.796078431372549 0.23529411764705882 1.0" />
    </material>




    <transmission name="jnt1_rev_simple_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="jnt1_rev">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="jnt1_rev_motor">
            <mechanicalReduction>1</mechanicalReduction>
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </actuator>
    </transmission>


    <transmission name="jnt2_rev_simple_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="jnt2_rev">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="jnt2_rev_motor">
            <mechanicalReduction>1</mechanicalReduction>
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </actuator>
    </transmission>


    <transmission name="jnt3_rev_simple_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="jnt3_rev">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="jnt3_rev_motor">
            <mechanicalReduction>1</mechanicalReduction>
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </actuator>
    </transmission>

    <transmission name="jnt4_rev_simple_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="jnt4_rev">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="jnt4_rev_motor">
            <mechanicalReduction>1</mechanicalReduction>
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </actuator>
    </transmission>

    <transmission name="jnt5_rev_simple_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="jnt5_rev">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="jnt5_rev_motor">
            <mechanicalReduction>1</mechanicalReduction>
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </actuator>
    </transmission>

    <transmission name="jnt6_rev_simple_trans">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="jnt6_rev">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="jnt6_rev_motor">
            <mechanicalReduction>1</mechanicalReduction>
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </actuator>
    </transmission>

    <gazebo>
        <plugin name="control" filename="libgazebo_ros_control.so">
            <robotNamespace>/</robotNamespace>  
        </plugin>
    </gazebo >

</robot>






